ext/standard/math.c

```diff
@@ -335,6 +335,17 @@ PHP_FUNCTION(round)
 		}
 	}
 
+	switch (mode) {
+		case PHP_ROUND_HALF_UP:
+		case PHP_ROUND_HALF_DOWN:
+		case PHP_ROUND_HALF_EVEN:
+		case PHP_ROUND_HALF_ODD:
+			break;
+		default:
+			zend_argument_value_error(3, "must be a valid rounding mode (PHP_ROUND_*)");
+			RETURN_THROWS();
+	}
+
```

Comment:
Previously an invalid rounding mode was considered to be equal to `PHP_ROUND_HALF_UP`. This is no longer the case since #12220. Making this a ValueError would be correct, but possibly a little steep. Shall I update this to a E_WARNING + change mode to PHP_ROUND_HALF_UP within the `switch()`? I don't want to update the rounding helper to handle unknown cases to keep the function clean.

Reply:
I think having it directly a ValueError is fine IMHO, just add an entry in UPGRADING